girder/cli/build.py

```diff
@@ -44,14 +44,18 @@
               help='Build girder library bundle in watch mode (implies --dev --no-reinstall).')
 @click.option('--watch-plugin',
               help='Build a girder plugin bundle in watch mode (implies --dev --no-reinstall).')
+@click.option('--npm',
+              help='Full path to the npm executable to use.')
 @click.option('--reinstall/--no-reinstall', default=True,
               help='Force regenerate node_modules.')
-def main(dev, watch, watch_plugin, reinstall):
-    npmCommand = os.getenv('NPM_EXE', 'npm')
-    if shutil.which(npmCommand) is None:
+def main(dev, watch, watch_plugin, npm, reinstall):
+    if npm is None:
+        npm = os.getenv('NPM_EXE', 'npm')
```

Comment:
If we do that, there is no way to detect if the user provided the flag or not.  Consider the following:
```bash
NPM_EXE='other-npm' girder build --npm npm
```
The usual convention would be to use `npm` in this case, but that wouldn't happen if we set the default in the decorator.

Reply:
Won't the below decorator achieve the behavior we want?

`@click.option('--npm', default=os.getenv('NPM_EXE', 'npm'))`

* If a `--npm` option is passed, it uses that.
* Otherwise, it looks at NPM_EXE env var
* If the env var isn't set, it uses `'npm'`

Am I misunderstanding how this works with click?

Reply:
No, my brain is broken this morning.